### DISCUSSION TITLE
where is the selection options in verion 4

### DISCUSSION THREAD

**Discussion by Hanisafa** [ME-MarvinE/XCalendar#56] [Q&A] [answered]
Hi , 
ive updated to v4 , 

SelectedDates not available , nor selection options  

how do i get the selected date/s in v4 ?

**Comment by ME-MarvinE** [comment, marked as answer]
Any properties to do with logic have been moved to the 'Calendar' class in XCalendar.Core. This is so you can inherit from and access the Calendar easier from code. It also makes it easier to create your own control to display the Calendar. You can find examples in the sample app.

```C#
using XCalendar.Core;
using XCalendar.Core.Models;
using XCalendar.Core.Enums;

public class MyClass
{
    public Calendar<CalendarDay> MyCalendar = new Calendar<CalendarDay>()
    {
        SelectionType = SelectionType.Single,
        SelectionAction = SelectionAction.Replace
    };

    public void UnSelectAllDates()
    {
        MyCalendar.SelectedDates.Clear();
    }
}
```

[Calendar Properties](https://github.com/ME-MarvinE/XCalendar/wiki/Calendar-Properties)
[CalendarView Properties](https://github.com/ME-MarvinE/XCalendar/wiki/CalendarView-Properties)
